outputs.tf

```diff
@@ -10,22 +11,28 @@ output "launch_config_outputs" {
   sensitive   = false
 }
 
-#TBD
+output "db_outputs" {
+  value       = var.deployment_type == "ec2" ? module.kong_ec2[0].db_outputs : null
+  description = "The DNS address and database name of the RDS instance, and security group ID from the database module."
+  sensitive   = false
+}
 
+## All
 output "private_subnet_azs" {
-  value       = var.deployment_type == "ec2" ? module.kong_ec2[0].private_subnet_azs : null
+  value       = var.deployment_type == "ec2" ? module.kong_ec2[0].private_subnet_azs : var.deployment_type == "ecs" ? module.kong_ecs[0].private_subnet_azs : null
   description = "List of availability zones used for the private subnets, either supplied in the optional `supplied in the optional `private_subnets` input variable or created in `subnets` submodule` input variable or defined in `subnets` submodule."
   sensitive   = false
 }
 
 output "private_subnet_ids" {
-  value       = var.deployment_type == "ec2" ? module.kong_ec2[0].private_subnet_ids : null
+  value       = var.deployment_type == "ec2" ? module.kong_ec2[0].private_subnet_ids : var.deployment_type == "ecs" ? module.kong_ecs[0].private_subnet_ids : null
   description = "List of private subnet IDs. These are either supplied in the optional `private_subnets` input variable or created in `subnets` submodule."
   sensitive   = false
 }
 
-output "db_outputs" {
-  value       = var.deployment_type == "ec2" ? module.kong_ec2[0].db_outputs : null
-  description = "The DNS address and database name of the RDS instance, and security group ID from the database module."
-  sensitive   = false
+## ECS
+
+output "kong_iam_role" {
```

Comment:
I suspect the EC2 deployment type also has a role that it can output here

Reply:
Can't see any applicable output, possibly not required for EC2

Reply:
the equivalent might be `module.kong_ec2[0].aws_launch_configuration.kong.iam_instance_profile`, but I'm not sure if that can be treated as a role

Reply:
The reason the IAM role is an output is so that policies can be attached to it, so I don't think in this instance `iam_instance_profile` would work - we could output the role thats attached to the iam instance profile in the ec2 module, but then thats arguably out of scope